modules/monitor/software/channel_monitor.cc

```diff
@@ -223,13 +223,16 @@ void ChannelMonitor::UpdateStatus(
 
   // Check channel frequency
   if (update_freq) {
-    if (freq > config.max_frequency_allowed()) {
+    if (config.has_max_frequency_allowed() &&
+        freq > config.max_frequency_allowed()) {
```

Comment:
Set a default value and just compare values, instead of checking `has_XXX()` first.